File: src/main/java/kr/co/lunatalk/global/util/SpringEnvironmentUtil.java
package kr.co.lunatalk.global.util;

import kr.co.lunatalk.global.common.constants.EnvironmentConstants;
import lombok.RequiredArgsConstructor;
import org.springframework.core.env.Environment;
import org.springframework.stereotype.Component;

import java.util.stream.Stream;

import static kr.co.lunatalk.global.common.constants.EnvironmentConstants.Constants.*;

@Component
@RequiredArgsConstructor
public class SpringEnvironmentUtil {

	private final Environment environment;

	public String getCurrentProfile() {
		return getActiveProfiles()
			.filter(profile -> profile.equals(PROD) || profile.equals(DEV))
			.findFirst()
			.orElse(LOCAL);
	}

	public boolean isProdProfile() {
		return getActiveProfiles().anyMatch(PROD::equals);
	}

	public boolean isDevProfile() {
		return getActiveProfiles().anyMatch(DEV::equals);
	}

	public boolean isLocalProfile() {
		return getActiveProfiles().anyMatch(LOCAL::equals);
	}

	private Stream<String> getActiveProfiles() {
		return Stream.of(environment.getActiveProfiles());
	}
}

File: lunatalk-common/src/main/kotlin/kr/co/lunatalk/global/util/SpringEnvironmentUtil.kt
package kr.co.lunatalk.global.util

import kr.co.lunatalk.global.common.constants.EnvironmentConstants.Constants.DEV
import kr.co.lunatalk.global.common.constants.EnvironmentConstants.Constants.LOCAL
import kr.co.lunatalk.global.common.constants.EnvironmentConstants.Constants.PROD
import org.springframework.core.env.Environment
import org.springframework.stereotype.Component

@Component
class SpringEnvironmentUtil(
    private val environment: Environment
) {

    fun getCurrentProfile(): String =
        getActiveProfiles()
            .filter { it == PROD || it == DEV }
            .firstOrNull()
            ?: LOCAL

    fun isProdProfile(): Boolean =
        getActiveProfiles().any { it == PROD }

    fun isDevProfile(): Boolean =
        getActiveProfiles().any { it == DEV }

    fun isLocalProfile(): Boolean =
        getActiveProfiles().any { it == LOCAL }

    private fun getActiveProfiles(): List<String> =
        environment.activeProfiles.toList()
}
